### PR TITLE
Feature/guest form submission

### PR DESCRIPTION
**Pull request by jasquat**
This adds in support to complete specified human tasks with a guest user.

To test:
1. Create a process model with a manual or user task and click the allow guest checkbox
1. Start the process model with the same user and go to the process instance show page to get the guid of the human task
1. Use that guid to create a url like: "[domain]/tasks/[process_instance_id]/[task_guid]" and navigate there from a browser that is not logged into spiff
1. This user should be able to complete the task and will be given the next allowed guest task if there is one
1. If the guest user tries to navigate anywhere that is not the task show page, they should be taken to the login screen

### CODE REVIEW

File: spiffworkflow-backend/src/spiffworkflow_backend/models/task.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ class TaskModel(SpiffworkflowBaseDBModel):
     human_tasks = relationship("HumanTaskModel", back_populates="task_model", cascade="delete")
     process_instance_id: int = db.Column(ForeignKey("process_instance.id"), nullable=False, index=True)
 
+    allow_guest: bool = db.Column(db.Boolean())
```
Comment (burnettk) [inline spiffworkflow-backend/src/spiffworkflow_backend/models/task.py:59]: could consider killing this cache column if the query filter isn't truly necessary.

File: spiffworkflow-backend/src/spiffworkflow_backend/services/user_service.py [hunk 2]
```diff
@@ -82,6 +83,14 @@ def current_user() -> Any:
             raise ApiError("logged_out", "You are no longer logged in.", status_code=401)
         return g.user
 
+    @classmethod
+    def is_logged_in_as_guest_user(cls) -> bool:
```
Comment (burnettk) [inline spiffworkflow-backend/src/spiffworkflow_backend/services/user_service.py:87]: i think we can kill this unless it's being used by something meta-programmed.

**Review by burnettk — APPROVED** [review]
it seems this one also needs the magic to combine the db migrations.

File: spiffworkflow-backend/src/spiffworkflow_backend/services/user_service.py (hunk 2 above)
Comment (jasquat) [reply spiffworkflow-backend/src/spiffworkflow_backend/services/user_service.py:87]: Looks like not. I think was using it but seems no longer necessary.

File: spiffworkflow-backend/src/spiffworkflow_backend/models/task.py (hunk 1 above)
Comment (jasquat) [reply spiffworkflow-backend/src/spiffworkflow_backend/models/task.py]: This has been removed.

**Pull request merged**